core/src/launcherworker.cpp

```diff
@@ -307,6 +308,15 @@ void LauncherWorker::update(
         return;
     }
 
+    AppInfo appInfo = api.getAppInfo(data.applicationSecret(), cancellationToken);
```

Comment:
I would suggest catching exceptions for it, since setting display name is not important functionality, yet it can break the launching process.